python/src/spark_rapids_ml/params.py

```diff
@@ -163,6 +176,20 @@ def num_workers(self) -> int:
     def num_workers(self, value: int) -> None:
         self._num_workers = value
 
+    def copy(self: P, extra: Optional["ParamMap"] = None) -> P:
+        # override this function to update cuml_params if possible
+        if isinstance(extra, dict):
+            for param, value in extra.items():
+                if isinstance(param, Param):
+                    self._set_cuml_param(param.name, value, silent=False)
```

Comment:
Should the extra items update the original or the copy?   Looks like it is updating the original here.

Reply:
Think this needs to update a copy.

Reply:
yes, it's a bug, I just fixed it.